torch/csrc/api/src/nn/modules/activation.cpp

```diff
@@ -390,5 +391,116 @@ void ThresholdImpl::pretty_print(std::ostream& stream) const {
   stream << ")";
 }
 
+// ============================================================================
+
+MultiheadAttentionImpl::MultiheadAttentionImpl(const MultiheadAttentionOptions& options_)
+    : options(options_) {
+  reset();
+}
+
+std::tuple<Tensor, Tensor> MultiheadAttentionImpl::forward(
+  const Tensor& query, const Tensor& key,
+  const Tensor& value, const Tensor& key_padding_mask,
+  bool need_weights, const Tensor& attn_mask) {
+  if (_qkv_same_embed_dim) {
```

Comment:
In the Python version, the first if-branch seems to be `if not self._qkv_same_embed_dim:`, is there a reason we switch the order?